handler.go

```diff
@@ -274,11 +274,11 @@ func (ca *Cagent) GetAllMeasurements() (MeasurementsMap, error) {
 	})
 
 	ca.hwInventory.Do(func() {
-		if hwInfo, err := hwinfo.Inventory(); err != nil && err != hwinfo.ErrNotPresent {
+		hwInfo, err := hwinfo.Inventory()
+		if err != nil {
 			errs = append(errs, err.Error())
-		} else if err == nil {
-			measurements = measurements.AddInnerWithPrefix("hw.inventory", hwInfo)
 		}
+		measurements = measurements.AddInnerWithPrefix("hw.inventory", hwInfo)
```

Comment:
We also add `hwInfo` in error case?

Reply:
Yes. This function can return partial result (as many others that are called in handler.GetMeasures()). For example, it can retrieve the list of USB devices but not the information about CPU (because the dmidecode is not installed). We should render as much info as possible in the result JSON, so we try. It's actually old code slightly refactored. I just made it more explicit.